server/container_create.go

```diff
@@ -1038,14 +1038,29 @@ func (s *Server) createSandboxContainer(ctx context.Context, containerID string,
 	}
 
 	// Get imageName and imageRef that are later requested in container status
-	status, err := s.StorageImageServer().ImageStatus(s.ImageContext(), images[0])
-	if err != nil {
-		return nil, err
+	var (
+		imgResult    *storage.ImageResult
+		imgResultErr error
+	)
+	if len(images) == 0 {
+		return nil, errors.New("cannot create a container when image is empty")
```

Comment:
Looks like we're getting this from `ResolveNames` above.  I think we should either shift this check inside `ResolveNames` (in our `pkg/storage/image.go`):

```go
  images := []string{}
  for _, r := range svc.registries {
    …
  }
  if len(images) == 0 {
    return nil, fmt.Errorf("could not resolve image name %q", imageName)
  }
  return images, nil
}
```

or make it part of the `ResolveNames` result checking here:

```go
images, err := s.StorageImageServer().ResolveNames(image)
if err != nil {
  …
}
if len(images) == 0 {
  return nil, fmt.Errorf("could not resolve image name %q", image)
}
```

Either way, it would be nice if `ResolveNames` had godocs explaining whether or not it considered “no resolved names” an error.

Reply:
That is what I was thinking too, but the purpose of ResolveNames is to get the full image path (either name or id) so I thought putting this in there was sort of taking away from what the function is actually meant to do.
My fix is finding which image actually exists in storage.
What do you think @mrunalp ?

Reply:
> …so I thought putting this in there was sort of taking away from what the function is actually meant to do.

That's fine with me.  Actually, with the [empty registries guard][1], I don't see a way that `images` could be empty.  Maybe add a comment [here][2] like:

```go
// Resolve names resolves an image name into a storage image ID or a fully-qualified
// image name (domain/repo/image).  This function will never return an empty array
// with a nil error. 
func (svc *imageService) ResolveNames(imageName string) ([]string, error) {
```

although that can certainly happen in follow-up work.

And in that case, we can drop the `len(images)` check entirely.

[1]: https://github.com/kubernetes-incubator/cri-o/blob/0b736bb43fb9f6ffa29b477be9e292bec3686ef4/pkg/storage/image.go#L643
[2]: https://github.com/kubernetes-incubator/cri-o/blob/0b736bb43fb9f6ffa29b477be9e292bec3686ef4/pkg/storage/image.go#L619

Reply:
Agreed, dropped the check and added the comment. thanks!